source/extensions/filters/common/lua/lua.cc

```diff
@@ -60,6 +68,11 @@ ThreadLocalState::ThreadLocalState(const std::string& code, ThreadLocal::SlotAll
 
   // Now initialize on all threads.
   tls_slot_->set([code](Event::Dispatcher&) {
+    // Avoid creating lua_State in different threads at the same time. In the process of creating
+    // lua_State, some static variables are used to assist memory allocation. Creating lua_State in
+    // different threads at the same time may cause data race. Although 'LuaJIT' uses retries to
+    // ensure the final safety of the memory allocation, it may still trigger the TSAN alarm.
+    Thread::LockGuard guard(luaThreadLocalLock());
```

Comment:
If we want to avoid the lock here, should we just disable the relevant tests under TSAN with a comment? I'm fine either way. cc @dio @lizan

Reply:
> If we want to avoid the lock here, should we just disable the relevant tests under TSAN with a comment? I'm fine either way. cc @dio @lizan

Yes. Only one test need to be disabled under TSAN. I think both would be acceptable, depending on what the community has to say.

Reply:
I'm good with that.

Reply:
Yeah let's just do that. Please have a good comment on why we are disabling TSAN for the test.

/wait

Reply:
ok 😄